src/components/core/GeoChart.tsx

```diff
@@ -44,6 +48,16 @@ export class GeoChartInner extends BaseVisualization<IGeoChartInnerProps, {}> {
         legendRenderer: renderLegend,
     };
 
+    public componentDidUpdate(prevProps: IGeoChartInnerProps) {
+        if (!isEqual(this.props.execution, prevProps.execution)) {
```

Comment:
I think we should put this checking to `shouldComponentUpdate`

Reply:
`validateData` function doesn’t stop the render flow. 
It only adds an `error` property into `BaseVisualization`.
`BaseVisualization` will use this `error` property to render the error screen